File: external/naver-client/src/main/java/com/news/feign/NaverClient.java
package com.news.feign;

import com.news.model.NaverWebResponse;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;

@FeignClient(name = "naverClient", url = "${external.naver.url}", configuration = NaverClientConfiguration.class)
public interface NaverClient {
    @GetMapping("/v1/search/webkr.json")
    NaverWebResponse search(@RequestParam("query") String query,
                            @RequestParam("start") int start,
                            @RequestParam("display") int display);
}

File: external/naver-client/src/main/kotlin/com/search/feign/NaverClient.kt
package com.search.feign

import com.search.config.feign.NaverClientConfiguration
import com.search.model.NaverWebResponse
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam

@FeignClient(name = "naverClient", url = "\${external.naver.url}", configuration = [NaverClientConfiguration::class])
interface NaverClient {
    @GetMapping("/v1/search/webkr.json")
    fun search(
            @RequestParam("query") query: String,
            @RequestParam("start") start: Int,
            @RequestParam("display") display: Int
    ) : NaverWebResponse
}